### DISCUSSION TITLE
Matrix with Relation fileds

### DISCUSSION THREAD

**Discussion by GeoSot** [orchidsoftware/platform#2199] [Q&A]
Hello guys, 

I am trying to learn the package, facing some issues (suppose based on documentation and my familiarity to this package)

Let's imagine we have a `TeamEditScreen` and want to create a `Layout` to list the team members combined with their roles (multiple)

Can anyone please help me hot to combine the proper fields in order to achieve this? 

The team relationships are based on jetstream implementation, where team has multiple users (rel:users) through an intermediate table (intermediate model: Membership ), pivoting roles

![image](https://user-images.githubusercontent.com/22406063/162232602-f5f11741-9dc1-4a10-8b9a-82400bc1f913.png)

Thank you in advance

**Comment by tabuna** [comment]
I think a table is a much better fit than a Matrix field in this case.

**Comment by GeoSot** [comment]
Please, if it is not much trouble , can you please provide me an example?

I am trying something like 


```php

                Matrix::make('team.users.')
                    ->title('Members list')
                    ->columns(['name', 'roles'])
                    ->fields([
                        'name' => Relation::make('team.users')
                            ->fromModel(User::class, 'name')
                            ->chunk(10),
                        'roles' => Select::make()->multiple()->fromModel(Role::class, 'name')->required(),
                    ]),

```

but the only thing can do, is to fill the right number of fields, with empty data

**Comment by AlizHarb** [comment]
I'm having same problem, getting empty data ,any soluation?

**Reply by AlizHarb** [reply]
easiest i found was to use each()
$user->load(['roles', 'items'])
            ->items->each(fn($item) => [$item->quantity = $item->pivot->quantity, $item->item_id = $item->pivot->item_id]);

might be helpful for someone :)
